jest.config.base.js

```diff
@@ -0,0 +1,27 @@
+const { defaults } = require("jest-config");
+
+module.exports = {
+  testEnvironment: "node",
+    setupFiles: [
+      "<rootDir>/../apollo-server-env/dist/index.js"
+    ],
+    preset: "ts-jest",
+    testMatch: null,
+    testRegex: "/__tests__/.*\\.test\\.(js|ts)$",
+    testPathIgnorePatterns: [
+      "/node_modules/",
+      "/dist/"
+    ],
+    moduleFileExtensions: [...defaults.moduleFileExtensions, "ts", "tsx"],
+    moduleNameMapper: {
+      '^__mocks__/(.*)$': '<rootDir>/../../__mocks__/$1',
+      '^(?!apollo-server-env|apollo-engine-reporting-protobuf)(apollo-(?:server|datasource|cache-control|tracing|engine)[^/]*|graphql-extensions)(?:/dist)?((?:/.*)|$)': '<rootDir>/../../packages/$1/src$2'
```

Comment:
This looks right to me — and I really like this approach since, as you said, it allows more correct mapping to the true source.  I assume the `/`s don't need to be escaped because they're passed to `new RegExp("...")`).  So I would probably say:

1. While re-using the `apollo-` prefix is nice, it makes refactoring a bit less clear.  For example, if I was changing the name of `apollo-cache-control` (or creating a new module based off it) a global grep on this repository for `apollo-cache-control` wouldn't bring me to this.  If we explicitly put `apollo-cache-control`, it would.  Though, I know that would make this very long.  I'm not sure what the best solution here is.
2. Since this is a `.js` file, maybe we could add a comment above about the intention here?  Just took me a bit to digest exactly the intion (not `apollo-server-env`, but some specific other packages was, and your explanation might get lost in this PR.
3. `engine` seems like it should be explicitly `engine-reporting` to not inadvertently match something else?